Managers/UInputMan.cpp

```diff
@@ -840,20 +840,25 @@ namespace RTE {
 		}
 
 		if (g_InActivity) {	
-			bool BuyMenuIsOpen = 0;
-			GameActivity* temp = dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity());
 
-			// Make ESC key open BuyMenuGUIs if any are open rather than pausing the game.
-			for (short player = Players::PlayerOne; player < Players::MaxPlayerCount; player++) {
-				if (temp->GetBuyGUI(player)->IsVisible()) {
-					BuyMenuIsOpen = true;
-					break;
+			if (AnyStartPress(false)) {
+
+				bool buyMenuIsOpen = 0;
+				GameActivity* currentGameActivity = dynamic_cast<GameActivity*>(g_ActivityMan.GetActivity());
+
+				// Make ESC key open BuyMenuGUIs if any are open rather than pausing the game.
```

Comment:
Sorry, I should have thought of this yesterday, but it was late and I was tired haha
Rather than have this loop and stuff in here, it'd be nicer to have a reusable method in game activity.
For maximum re-usability, I would suggest adding something like `BuyMenuIsOpen(int playerNumber)`. This method would return whether the given player's buy menu is open or not. Passing in -1 would iterate through all current players, and tell you if any of their buy menus are open (this is fine since player -1 would never have a buy menu).